cw_decred/lib/wallet_service.dart

```diff
@@ -57,42 +58,89 @@ class DecredWalletService extends WalletService<
+    // is not used. An older wallet will have a directory here which is a
+    // condition for moving the wallet when opening, so this must be kept blank
+    // going forward.
+    credentials.walletInfo!.dirPath = "";
+    credentials.walletInfo!.path = "";
     final wallet = DecredWallet(credentials.walletInfo!, credentials.password!,
         this.unspentCoinsInfoSource, libwallet!, closeLibwallet);
     await wallet.init();
     return wallet;
   }
 
+  void copyDirectorySync(Directory source, Directory destination) {
+    /// create destination folder if not exist
+    if (!destination.existsSync()) {
+      destination.createSync(recursive: true);
+    }
+
+    /// get all files from source (recursive: false is important here)
+    source.listSync(recursive: false).forEach((entity) {
+      final newPath = destination.path + Platform.pathSeparator + basename(entity.path);
+      if (entity is File) {
+        entity.rename(newPath);
+      } else if (entity is Directory) {
+        copyDirectorySync(entity, Directory(newPath));
+      }
+    });
+  }
+
+  Future<void> moveWallet(String fromPath, String toPath) async {
+    final oldWalletDir = new Directory(fromPath);
+    final newWalletDir = new Directory(toPath);
+    copyDirectorySync(oldWalletDir, newWalletDir);
+    await oldWalletDir.delete(recursive: true);
+  }
+
   @override
   Future<DecredWallet> openWallet(String name, String password) async {
     final walletInfo = walletInfoSource.values
         .firstWhereOrNull((info) => info.id == WalletBase.idFor(name, getType()))!;
-    final network = walletInfo.derivationInfo?.derivationPath == seedRestorePathTestnet ||
-            walletInfo.derivationInfo?.derivationPath == pubkeyRestorePathTestnet
-        ? testnet
-        : mainnet;
+    if (walletInfo.network == null || walletInfo.network == "") {
+      walletInfo.network = walletInfo.derivationInfo?.derivationPath == seedRestorePathTestnet ||
+              walletInfo.derivationInfo?.derivationPath == pubkeyRestorePathTestnet
+          ? testnet
+          : mainnet;
+    }
 
     await this.init();
-    final walletDirExists = Directory(walletInfo.dirPath).existsSync();
-    if (!walletDirExists) {
-      walletInfo.dirPath = await pathForWalletDir(name: name, type: getType());
+
+    // Cake wallet version 4.27.0 and earlier gave a wallet dir that did not
+    // match the name. Move those to the correct place.
+    final dirPath = await pathForWalletDir(name: name, type: getType());
+    if (walletInfo.dirPath != "") {
+      // On ios the stored dir no longer exists. We can only trust the basename.
+      final randomBasename = basename(walletInfo.dirPath);
+      final oldDir = await pathForWalletDir(name: randomBasename, type: getType());
+      if (oldDir != dirPath) {
+        await this.moveWallet(oldDir, dirPath);
+      }
+      // Clear the path so this does not trigger again.
+      walletInfo.dirPath = "";
+      walletInfo.path = "";
```

Comment:
We should call `walletInfo.save()`
```suggestion
      walletInfo.dirPath = "";
      walletInfo.path = "";
      walletInfo.save();
```